bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt

```diff
@@ -103,94 +104,120 @@ internal fun newGoalEvent(
   )
 }
 
-internal fun newGetEvaluationLatencyMetricsEvent(
+internal fun newMetricsEvent(
   clock: Clock,
   idGenerator: IdGenerator,
-  seconds: Long,
-  featureTag: String,
+  featureTag: String?,
   appVersion: String,
+  metricsEventType: MetricsEventType,
+  apiID: ApiID,
+  // note: only available on success request
+  latencySecond: Long? = null,
+  // note: only available on success request
+  sizeByte: Int? = null,
 ): Event {
   return Event(
     id = idGenerator.newId(),
     type = EventType.METRICS,
     event = EventData.MetricsEvent(
       timestamp = clock.currentTimeSeconds(),
-      type = MetricsEventType.GET_EVALUATION_LATENCY,
-      event = MetricsEventData.GetEvaluationLatencyMetricsEvent(
-        labels = mapOf(
-          "tag" to featureTag,
-        ),
-        duration = seconds,
-      ),
+      type = metricsEventType,
+      event = newMetricsEventData(featureTag, apiID, metricsEventType, latencySecond, sizeByte),
       sdkVersion = BuildConfig.SDK_VERSION,
       metadata = newMetadata(appVersion),
     ),
   )
 }
+internal fun newMetricsEventData(
+  featureTag: String?,
+  apiID: ApiID,
+  type: MetricsEventType,
+  // note: only available on success request
+  latencySecond: Long? = null,
```

Comment:
I think it will be better if we split the function or refactor it. One for success and another for failure, so we can avoid those checking for latencySecond and sizeByte throwing an exception.

Reply:
its ready now